src/data/list/big_operators.lean

```diff
@@ -482,19 +484,52 @@ lemma tail_sum (L : list ℕ) : L.tail.sum = L.sum - L.head :=
 by rw [← head_add_tail_sum L, add_comm, add_tsub_cancel_right]
 
 section alternating
-variables [comm_group G]
+section
+variables [has_one α] [has_mul α] [has_inv α]
+
+@[simp, to_additive] lemma alternating_prod_nil : alternating_prod ([] : list α) = 1 := rfl
+@[simp, to_additive] lemma alternating_prod_singleton (a : α) : alternating_prod [a] = a := rfl
+
+@[to_additive] lemma alternating_prod_cons_cons' (a b : α) (l : list α) :
+  alternating_prod (a :: b :: l) = a * b⁻¹ * alternating_prod l := rfl
+
+end
+
+@[to_additive] lemma alternating_prod_cons_cons [group α] (a b : α) (l : list α) :
```

Comment:
If you're squeezing typeclass assumptions, then you can use `div_inv_monoid` here.